src/backend/api/Fusion.Resources.Api/Controllers/Utilities/UtilitiesController.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Fusion.Integration.Configuration;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Api.Controllers.Utilities
+{
+    [Authorize]
+    [ApiController]
+    public class UtilitiesController : ResourceControllerBase
+    {
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly IFusionTokenProvider tokenProvider;
+        private readonly IConfiguration configuration;
+        private readonly IOptions<FusionIntegrationOptions> fusionOptions;
+
+        public UtilitiesController(IHttpClientFactory httpClientFactory, IFusionTokenProvider tokenProvider, IConfiguration configuration, IOptions<FusionIntegrationOptions> fusionOptions)
+        {
+            this.httpClientFactory = httpClientFactory;
+            this.tokenProvider = tokenProvider;
+            this.configuration = configuration;
+            this.fusionOptions = fusionOptions;
+        }
+
+        [HttpPost("/utilities/parse-spreadsheet")]
+        public async Task<ActionResult<ExcelConversion>> ValidateContractorImportSpreadsheet([FromForm]ConvertSpreadsheetRequest request)
+        {
+            if (request == null)
+                return FusionApiError.InvalidOperation("MissingBody", "Could not locate any body payload");
+
+            var url = $"https://pro-f-utility-{fusionOptions.Value.Environment}.azurewebsites.net";
+            var token = await tokenProvider.GetApplicationTokenAsync();
+            var client = httpClientFactory.CreateClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", token);
+            client.BaseAddress = new Uri(url);
+
+
+            if (string.IsNullOrEmpty(url))
+                throw new InvalidOperationException("Missing configuration for fusion utility function");
```

Comment:
Is there a specific reason why is this not also using FusionApiError.InvalidOperation(...) rather than throwing exception directly?

Reply:
Because it is not a user error, it is a configuration error, 500 not 400.

Reply:
Ok - agree in regards to status code. We could still do FusionApiError.Internal(...)  with the same API model as FusionApiError.InvalidOperation(), but it's not a big deal.